pkg/operator/observe_config.go

```diff
@@ -120,6 +127,28 @@ func observeControllerManagerImagesConfig(kubeClient kubernetes.Interface, confi
 	return observedConfig, nil
 }
 
+// observeInternalRegistryHostname reads the internal registry hostname from the cluster configuration as provided by
+// the registry operator.
+func observeInternalRegistryHostname(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	imageConfigClient, err := configv1client.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	configImage, err := imageConfigClient.Images().Get("cluster", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return observedConfig, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	internalRegistryHostName := configImage.Status.InternalRegistryHostname
+	if len(internalRegistryHostName) > 0 {
+		unstructured.SetNestedField(observedConfig, internalRegistryHostName, "imagePolicyConfig", "internalRegistryHostname")
```

Comment:
@bparees iirc the observedConfig is an `OpenshiftControllerManagerConfig` object, so the copy-paste from @juanvallejo won't fully translate here.

Reply:
I would expect controllermanagerconfig and api config to contain the same configuration information (an equivalent to the master-config.yaml that both of them consumed pre-4.0).

But @deads2k can weigh in

Reply:
it is a struct of this information: https://github.com/openshift/api/blob/master/openshiftcontrolplane/v1/types.go#L141

the API types are different because the binaries are different and require different information.

Reply:
ok i've fixed this, but holding pending creation of tests.